### PR TITLE
Allows commas to separate cookies in RFC2965 compliance mode

### PR DESCRIPTION
**Pull request by gregw**
For #3041 allow commas as a cookie separator in RFC2965 compliance mode.

### CODE REVIEW

**Comment by gregw** [comment]
@joakime actually, I think we need separate compliance modes for sending and receiving cookies.   Just because we are prepared to accepted RFC2965 cookies does not necessarily mean we should generate set-cookies like that!    Should we fix that as part of this issue?

File: jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java [hunk 1]
```diff
@@ -269,34 +316,19 @@ else if(tokenstart>=0 && tokenend>=0)
...
-                                else if(tokenstart>=0 && tokenend>=0)
-                                {
-                                    name = hdr.substring(tokenstart, tokenend+1);
-                                }
+                                else if(tokenstart>=0)
+                                    name = tokenend>=tokenstart?hdr.substring(tokenstart, tokenend+1):hdr.substring(tokenstart);
+
                                 tokenstart = -1;
-                                invalue=true;
+                                if (c=='=')
```
Comment (jborik5) [inline jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java]: Why is this if needed? It seems redundant to me.

Comment (gregw) [reply jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java]: Ah true... at one stage I was handling '0' and ';' name termination... but no longer needed

File: jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java [hunk 2]
```diff
@@ -170,6 +170,7 @@ public SessionAsyncListener(Session session)
         public void onComplete(AsyncEvent event) throws IOException
         {
             //An async request has completed, so we can complete the session
+            complete(Request.getBaseRequest(event.getAsyncContext().getRequest()).getSession(false));
```
Comment (joakime) [inline jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java]: This isn't relevant to this issue.

**Comment by joakime** [comment]
@gregw Is the CookieCutter used for jetty-client too? (thinking about parsing of `Set-Cookie` from bad servers?)

Also, I think we (Jetty) should never generate a Cookie with commas.
But I think that should be a separate issue.

**Pull request merged**